### ISSUE TITLE
Output isn't line buffered

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/compiler-explorer#2301]
Probably as we redirect output to a pipe, anything that presumes buffers flushed on `\n`...are not diusplayed if the app crashes. This makes for unfortunate debugging:

https://godbolt.org/z/WooxnP

I /think/ there's a way to force output to be line buffered somehow?

**Comment by partouf** [comment]
I think I remember this behaviour being different when ran locally without nsjail, but not actually sure.

**Comment by apmorton** [comment]
This is just "one of those things".
The behavior does not differ if run locally without any sandboxing.

However, the same binary run from a terminal will correctly output before terminating.

Adding `std::setbuf(stdout, NULL)` to the top of main "fixes" it in CE.

This is because:
>Normally all files are block buffered.  If a stream refers to a
> terminal (as stdout normally does), it is line buffered.  The
> standard error stream stderr is always unbuffered by default.

https://man7.org/linux/man-pages/man3/setbuf.3.html

Because we redirect all IO to pipes, this automatic line buffering does not happen, and stdout defaults to block buffered.

Short of executing with a [fake tty](https://github.com/microsoft/node-pty) I don't know if there is a way to fix this.

**Comment by mattgodbolt** [comment]
Right, faketty was what I was thinking though. It is annoying that if an app crashes any output it was writing is lost. This came up in a discussion about some assertions in libc itself which sadly never show up as they do printf("oh noes your broke it\n"); abort();

**Comment by mattgodbolt** [comment]
Done; thanks @OfekShilon (and @narpfel for pointing this out)
